pkgs/development/interpreters/python/cpython/default.nix

```diff
@@ -262,7 +262,7 @@ in with passthru; stdenv.mkDerivation {
   # Enforce that we don't have references to the OpenSSL -dev package, which we
   # explicitly specify in our configure flags above.
   disallowedReferences =
-    stdenv.lib.optionals (openssl != null) [ openssl.dev ]
+    stdenv.lib.optionals (openssl != null && !stdenv.hostPlatform.isMusl) [ openssl.dev ]
```

Comment:
I don't like this. It shouldn't be done just when using Musl, but when performing static builds. We need a way to check whether we're creating static builds.

Reply:
see https://github.com/NixOS/nixpkgs/issues/83664